apps/files_sharing/lib/Controller/ShareesAPIController.php

```diff
@@ -197,6 +197,27 @@ public function search(string $search = '', ?string $itemType = null, int $page
 			$result['exact'] = array_merge($this->result['exact'], $result['exact']);
 		}
 		$this->result = array_merge($this->result, $result);
+
+		// Get a list of groups to exclude from search results
+		$blacklisted_groups = $this->config->getSystemValue('blacklisted_groups', true);
```

Comment:
```suggestion
		// Get a list of groups to exclude from search results
		$sharingDisabledGroups = $this->config->getSystemValue('sharing.disabled_groups', true);
```

This name is too generic for a system option and `blacklist` should be avoided (it's better and clearer to use `blocklist`).